cloudweatherreport/cloudresource/resource.py

```diff
@@ -0,0 +1,153 @@
+import logging
+import os
+import yaml
+
+import cloudweatherreport.cloudresource.aws as aws
+import cloudweatherreport.cloudresource.gce as gce
+import cloudweatherreport.cloudresource.azure as azure
+from cloudweatherreport.utils import juju_cmd
+
+
+class UnknownCloudName(Exception):
+    pass
+
+
+class UnknownCredentialName(Exception):
+    pass
+
+
+class CloudNotSupported(Exception):
+    pass
+
+
+def get_credential_name(creds, cloud_name):
+    cred_name = creds.get('credentials', {}).get(cloud_name, {}).get(
+        'default-credential')
+    if cred_name:
+        return cred_name
+    # If no default credential name, get the first name from the list
+    names = list(creds.get('credentials', {}).get(cloud_name, {}).keys())
```

Comment:
Thank you for catching this. According to this doc, I only see `default-region` and `default-credential` 
https://jujucharms.com/docs/2.0/credentials

I think you right about we should follow what Juju does. If there are multiple credential names and no default, we should raise an exception.